### ISSUE TITLE
Many warnings "has a resource with the locale 'de'. This locale is not recognized by .NET"

### ISSUE THREAD

**Issue by jakubch1** [dotnet/sdk#37636]
Our CI build is producing many warnings like this:
```
/__w/1/s/.dotnet/sdk/9.0.100-alpha.1.23615.4/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(266,5): warning NETSDK1188: Package Microsoft.Testing.Platform 1.0.0-preview.23621.6 has a resource with the locale 'cs'. This locale is not recognized by .NET. Consider notifying the package author that it appears to be using an invalid locale. [/__w/1/s/tests/Microsoft.CodeCoverage.Analysis.Tests/Microsoft.CodeCoverage.Analysis.Tests.csproj::TargetFramework=net9.0]
##[warning].dotnet/sdk/9.0.100-alpha.1.23615.4/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(266,5): warning NETSDK1188: (NETCORE_ENGINEERING_TELEMETRY=Build) Package Microsoft.Testing.Platform 1.0.0-preview.23621.6 has a resource with the locale 'cs'. This locale is not recognized by .NET. Consider notifying the package author that it appears to be using an invalid locale.
/__w/1/s/.dotnet/sdk/9.0.100-alpha.1.23615.4/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(266,5): warning NETSDK1188: Package Microsoft.Testing.Platform 1.0.0-preview.23621.6 has a resource with the locale 'de'. This locale is not recognized by .NET. Consider notifying the package author that it appears to be using an invalid locale. [/__w/1/s/tests/Microsoft.CodeCoverage.Analysis.Tests/Microsoft.CodeCoverage.Analysis.Tests.csproj::TargetFramework=net9.0]
##[warning].dotnet/sdk/9.0.100-alpha.1.23615.4/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(266,5): warning NETSDK1188: (NETCORE_ENGINEERING_TELEMETRY=Build) Package Microsoft.Testing.Platform 1.0.0-preview.23621.6 has a resource with the locale 'de'. This locale is not recognized by .NET. Consider notifying the package author that it appears to be using an invalid locale.
```

It is happening only on unix (ubuntu, alpine and macos). Building same code on windows is not causing it. I am also not able to repro it locally. We build our code using arcade infrastructure. Link to example build: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=8840677&view=logs&j=9da81d86-a7e8-53ea-404c-b4879343d50c&t=9ef44dce-4b34-5e1a-8b43-8a202ad719eb&l=77

**Comment by jakubch1** [comment]
When building we were setting env var `DOTNET_SYSTEM_GLOBALIZATION_INVARIANT` to `1`. This was causing all above warnings. Probably fix should check for this env var and not generate warnings if set to `1`.

**Comment by ihordyrman** [comment]
We're running into the exact same issue in our solution. We're also using DOTNET_SYSTEM_GLOBALIZATION_INVARIANT, but still getting a ton of these NETSDK1188 warnings. Here's an example from our logs:

>[WRN] /home/gitlab-runner/builds/g_czk3Jmq/0/*/.nuke/temp/dotnet-unix/sdk/8.0.200/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(266,5): warning NETSDK1188: Package Microsoft.CodeAnalysis.CSharp 4.5.0 has a resource with the locale 'pl'. This locale is not recognized by .NET. Consider notifying the package author that it appears to be using an invalid locale. [/home/gitlab-runner/builds/g_czk3Jmq/0/*.csproj]

Tried the global <NoWarn>$(NoWarn);NETSDK1188</NoWarn>, but no luck – the warnings are still there and cluttering up our logs.

**Comment by jorgelaranjo** [comment]
`export DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=0
`

Worth checking documentation
https://learn.microsoft.com/en-us/dotnet/core/runtime-config/globalization

**Comment by freddyrios** [comment]
Installing globalization support is not a solution but a workaround.

Specially as we are talking about test platform packagesa. Why would it force you to change the environment to avoid warnings about cultures that are not relevant for the test environment.

**Comment by iq2luc** [comment]
> Installing&enabling globalization support is not a solution but a workaround.

My "solution" is `<NoWarn>NETSDK1188</NoWarn>`. :-)

**Comment by codepic** [comment]
> My "solution" is `<NoWarn>NETSDK1188</NoWarn>`. :-)

Perhaps even `<NoWarn>$(NoWarn);NETSDK1188</NoWarn>` to append to any existing warnings being suppressed.

**Comment by iq2luc** [comment]
> > My "solution" is `<NoWarn>NETSDK1188</NoWarn>`. :-)
> 
> Perhaps even `<NoWarn>$(NoWarn);NETSDK1188</NoWarn>` to append to any existing warnings being suppressed.

Absolutely, I initially posted it more ore less like a joke, but that's exactly what I ended up doing (your version to be more precise).

**Comment by andrewlock** [comment]
FYI, I solved this recently by [disabling satellite assemblies](https://andrewlock.net/disabling-localized-satellite-assemblies-during-dotnet-publish/#disabling-copying-of-satellite-assemblies-during-publish) in the test projects by setting 

```xml
<SatelliteResourceLanguages>en</SatelliteResourceLanguages>
```

No more warnings after that 🙂

**Comment by JuanZamudioGBM** [comment]
> FYI, I solved this recently by [disabling satellite assemblies](https://andrewlock.net/disabling-localized-satellite-assemblies-during-dotnet-publish/#disabling-copying-of-satellite-assemblies-during-publish) in the test projects by setting
> 
> <SatelliteResourceLanguages>en</SatelliteResourceLanguages>
> No more warnings after that 🙂

I have that in my unit tests, but I still see the warning in a Linux container running on arm64

**Comment by marcpopMSFT** [comment]
Old issue triage: Sounds like a workaround was found. These are valid locales I think so not sure why it's happening but closing for now.
